File: R/basque.R
require(Synth)

data(basque)

treated.regions <- 'Basque Country (Pais Vasco)'
control.regions <- c('Spain (Espana)', 'Andalucia', 'Aragon',
                     'Principado De Asturias', 'Baleares (Islas)', 'Canarias',                   
                     'Cantabria', 'Castilla Y Leon', 'Castilla-La Mancha',
                     'Cataluna', 'Comunidad Valenciana', 'Extremadura',
                     'Galicia', 'Madrid (Comunidad De)', 'Murcia (Region de)',
                     'Navarra (Comunidad Foral De)', 'Rioja (La)')

dataprep.out <- dataprep(
  foo = basque,
  predictors = c('school.illit', 'school.prim', 'school.med',
                 'school.high', 'school.post.high', 'invest'),
  predictors.op = 'mean',
  time.predictors.prior = 1964:1969,
  special.predictors = list(
    list('gdpcap', 1960:1969 ,'mean'),
    list('sec.agriculture', seq(1961, 1969, 2), 'mean'),
    list('sec.energy', seq(1961, 1969, 2), 'mean'),
    list('sec.industry', seq(1961, 1969, 2), 'mean'),
    list('sec.construction', seq(1961, 1969, 2), 'mean'),
    list('sec.services.venta', seq(1961, 1969, 2), 'mean'),
    list('sec.services.nonventa', seq(1961, 1969, 2), 'mean'),
    list('popdens', 1969, 'mean')
  ),
  dependent = 'gdpcap',
  unit.variable = 'regionno',
  unit.names.variable = 'regionname',
  time.variable = 'year',
  treatment.identifier = 17,
  controls.identifier = control.regions,
  time.optimize.ssr = 1960:1969,
  time.plot = 1955:1997
)

synth.out <- synth(
  data.prep.obj = dataprep.out,
  method = 'BFGS'
)

path.plot(
  dataprep.res = dataprep.out,
  synth.res = synth.out,
  tr.intake = 1998
)

state.weights <- setNames(
  control.regions,
  round(synth.out$solution.w, digits = 2)
)
print(state.weights)

File: examples/basque.py
import pandas as pd
from ..pyaugsynth import dataprep


df = pd.read_csv("../data/basque.csv")

out = dataprep(
    foo=df,
    predictors=(
        "school.illit",
        "school.prim",
        "school.med",
        "school.high",
        "school.post.high",
        "invest",
    ),
    predictors_op="mean",
    time_predictors_prior=range(1964, 1970),
    special_predictors=(
        ("gdpcap", range(1960, 1970), "mean"),
        ("sec.agriculture", range(1961, 1970, 2), "mean"),
        ("sec.energy", range(1961, 1970, 2), "mean"),
        ("sec.industry", range(1961, 1970, 2), "mean"),
        ("sec.construction", range(1961, 1970, 2), "mean"),
        ("sec.services.venta", range(1961, 1970, 2), "mean"),
        ("sec.services.nonventa", range(1961, 1970, 2), "mean"),
        ("popdens", [1969], "mean"),
    ),
    dependent="gdpcap",
    unit_variable="regionname",
    time_variable="year",
    treatment_identifier="Basque Country (Pais Vasco)",
    controls_identifier=(
        "Spain (Espana)",
        "Andalucia",
        "Aragon",
        "Principado De Asturias",
        "Baleares (Islas)",
        "Canarias",
        "Cantabria",
        "Castilla Y Leon",
        "Castilla-La Mancha",
        "Cataluna",
        "Comunidad Valenciana",
        "Extremadura",
        "Galicia",
        "Madrid (Comunidad De)",
        "Murcia (Region de)",
        "Navarra (Comunidad Foral De)",
        "Rioja (La)",
    ),
    time_optimize_ssr=range(1960, 1970),
    time_plot=range(1955, 1998),
)
